FE/angular/src/app/components/movie/movie.component.ts

```diff
@@ -1,25 +1,22 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { tap } from 'rxjs';
+import { Observable, switchMap } from 'rxjs';
 import { DataService, MovieComplete } from '../../services/data.service';
 
 @Component({
   selector: 'app-movie',
   templateUrl: './movie.component.html'
 })
-export class MovieComponent implements OnDestroy, OnInit {
-  public movie: MovieComplete;
-  public movieId = '';
-  private movieSubscription: any;
+export class MovieComponent implements OnInit {
+  public movieSubscription: Observable<MovieComplete>;
 
   constructor(private activatedRoute: ActivatedRoute, private dataService: DataService) {}
 
   public ngOnInit() {
```

Comment:
Do you need `ngOnInit()`?
Please explain why or not.

Reply:
ngOnInit() is the place where we should read the state/router data.